### PR TITLE
Sim agent faces performer during interaction

### PR DESCRIPTION
**Pull request by jaudick**
[MCS](https://github.com/NextCenturyCorporation/MCS/pull/549)
You need the mini agents from [1257](https://github.com/NextCenturyCorporation/mcs-private/pull/176)

### CODE REVIEW

**Comment by jaudick** [comment]
@kdrumm I made it intentional that agents holding a ball out for you will track you until you pick it up, while agents not holding a ball only rotate to you during their 5 frame sad interaction animation.

**Review by kdrumm — APPROVED** [review]

**Review by brianpippin — APPROVED** [review]

File: unity/Assets/Scripts/MCSSimulationAgent.cs [hunk 1]
```diff
@@ -186,31 +199,93 @@ public enum SimAgentActionState {
...
+        float target = targetRotation.y;
+        float original = originalRotation.y;
+        float degreeChange = 0;
+        
+        if(target > original + 180) {
+            target -= 360;
+        }
+        else if(target < original - 179) {
+            target += 360;
+        }
```
Comment (ThomasSchellenbergNextCentury) [inline unity/Assets/Scripts/MCSSimulationAgent.cs:245]: Are we concerned about rotation values being greater than 360 or less than 0? Or are we already handling that case somewhere?

Comment (jaudick) [reply unity/Assets/Scripts/MCSSimulationAgent.cs:245]: This code will also handle that scenario too, I just tested rot<0 and rot>360 scenarios right now and it worked

Comment (ThomasSchellenbergNextCentury) [reply unity/Assets/Scripts/MCSSimulationAgent.cs:245]: Awesome! Thanks for checking.

**Review by ThomasSchellenbergNextCentury — APPROVED** [review]

**Pull request merged**